src/bilstm_train_regression.py

```diff
@@ -0,0 +1,256 @@
+
+
+if __name__ == '__main__':
+
+    args = get_args()
+    logger.info(f"Reading test file")
+    test = process_data_regression(args.input_path, args.test_filename)
+    X_test = test.drop(args.factor, axis=1)
+    texts_test = X_test["Text-EN"].str.lower()
+    cards_test = X_test.filter(regex="card_.*")  # Using a regular expression cause some cards may not be in test.
+    y_test = test[args.factor]
+    logger.info(f" Using test dataset with {X_test.shape[0]} instances")
+
+    logger.info(f"Reading train file")
+    train = process_data_regression(args.input_path, args.train_filename)
+    X_train = train.drop(args.factor, axis=1)
+    texts_train = X_train["Text-EN"].str.lower()
+    cards_train = X_train.filter(regex="card_.*")  # Using a regular expression cause some cards may not be in train.
+    y_train = train[args.factor]
+    y_train_norm = (y_train - y_train.mean())/y_train.std()
+    y_test_norm = (y_test - y_train.mean())/y_train.std()
+    logger.info(f" Using train dataset with {X_train.shape[0]} instances")
+
+    tok = get_tokenizer(args)
+    tok.fit_on_texts(list(texts_train) + list(texts_test))
+    texts_train = tok.texts_to_sequences(texts_train)
+    texts_test = tok.texts_to_sequences(texts_test)
+    x_train = sequence.pad_sequences(texts_train, maxlen=args.max_length)
+    x_test = sequence.pad_sequences(texts_test, maxlen=args.max_length)
+
+    embeddings_index = {}
+    with open(os.path.join(args.embedding_path, args.embedding_file), encoding='utf8') as f:
+        for line in f:
+            values = line.rstrip().rsplit(' ')
+            word = values[0]
+            coefficients = np.asarray(values[1:], dtype='float32')
+            embeddings_index[word] = coefficients
+
+    word_index = tok.word_index
+    # prepare embedding matrix
+    num_words = min(args.max_features, len(word_index) + 1)
+    embedding_matrix = np.zeros((num_words, args.embed_size))
+    for word, i in word_index.items():
+        if i >= args.max_features:
+            continue
+        embedding_vector = embeddings_index.get(word)
+        if embedding_vector is not None:
+            # words not found in embedding index will be all-zeros.
+            embedding_matrix[i] = embedding_vector
+
+    text_input = Input(shape=(args.max_length,), name='text_input')
+    x = Embedding(args.max_features, args.embed_size, weights=[embedding_matrix], trainable=False)(text_input)
+    x = SpatialDropout1D(0.2)(x)
+
+    # Adding context
+    auxiliary_input = Input(shape=(len(cards_train.columns),), name='aux_input')
+    x_1 = RepeatVector(args.max_length)(auxiliary_input)
+    x = keras.layers.concatenate([x, x_1])
+
+    x = Bidirectional(GRU(128, return_sequences=True, dropout=0.1, recurrent_dropout=0.1))(x)
+    x = Conv1D(64, kernel_size=3, padding="valid", kernel_initializer="glorot_uniform")(x)
+    avg_pool = GlobalAveragePooling1D()(x)
+    max_pool = GlobalMaxPooling1D()(x)
+    x = concatenate([avg_pool, max_pool])
```

Comment:
It is better to chose either objects of functions instead of mixing them both in the same program. I would prefer the object concatenate here as the rest of the elements are objects. It is easier to understand if everything is consistent.